### PR TITLE
[JENKINS-63332] Radio button's move around when clicking them

### PR DESCRIPTION
**Pull request by benebsiny**
See [JENKINS-63332](https://issues.jenkins-ci.org/browse/JENKINS-63332).

I set 'visibility' to 'hidden' and 'position' to 'absolute' to hide the foldable block that relative to the radio button that was not clicked by user.
### Proposed changelog entries

* JENKINS-63332: Fix bug that radio button move around when clicking them.

### Proposed upgrade guidelines

N/A

### Submitter checklist

- [X] (If applicable) Jira issue is well described
- [X] Changelog entries and upgrade guidelines are appropriate for the audience affected by the change (users or developer, depending on the change). [Examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)
  * Fill-in the `Proposed changelog entries` section only if there are breaking changes or other changes which may require extra steps from users during the upgrade
- [X] Appropriate autotests or explanation to why this change has no tests
- ~~[ ] For dependency updates: links to external changelogs and, if possible, full diffs~~

### Desired reviewers

@mention

### Maintainer checklist

Before the changes are marked as `ready-for-merge`: 

- [ ] There are at least 2 approvals for the pull request and no outstanding requests for change
- [ ] Conversations in the pull request are over OR it is explicit that a reviewer does not block the change
- [ ] Changelog entries in the PR title and/or `Proposed changelog entries` are correct
- [ ] Proper changelog labels are set so that the changelog can be generated automatically
- [ ] If the change needs additional upgrade steps from users, `upgrade-guide-needed` label is set and there is a `Proposed upgrade guidelines` section in the PR title. ([example](https://github.com/jenkinsci/jenkins/pull/4387))
- [ ] If it would make sense to backport the change to LTS, a Jira issue must exist, be a _Bug_ or _Improvement_, and be labeled as `lts-candidate` to be considered (see [query](https://issues.jenkins-ci.org/issues/?filter=12146)).

### CODE REVIEW

**Comment by benebsiny** [comment]
Before
i
After
o

**Review by timja — APPROVED** [review]
Thanks! Looks fine to me, @fqueiruga ?

**Review by oleg-nenashev — APPROVED** [review]
LGTM, thanks for the patch! I am not very familiar with these styles, but looks legit from the documentation

**Review by varyvol — APPROVED** [review]

**Review by alecharp — APPROVED** [review]
LGTM. thanks!

**Comment by alecharp** [comment]
This PR is now ready for merge, after ~24 hours, we will merge it if there's no negative feedback.

**Review by romenrg — COMMENTED** [review]
Wow. First time I see the radioblock code and dig into its behavior. The whole thing seems very "hacky". It apparently adds a few table rows per option and, when a new radio button is selected, it changes the visibility of an almost-still-invisible row of those, just causing a small jump because the "empty" element has a 2px height.

The proposed fix seems to fix that jump, removing the immediate annoying visual issue, but it does seem we should think of changing the whole behavior of this piece.

Only thing to note regarding the current proposed fix is that AFAIK, the default value for "position" is "static". Setting it to "" (empty string) means browsers probably will detect an invalid value and likely default to "static"; so I guess not a big deal, but I would seem to me more correct to use 'static' instead of "".

Apart from all that, thanks a lot @benebsiny for the fix!

---

_**EDIT**: I've removed my original suggestion, as I've found a better alternative [described in my latest comment](https://github.com/jenkinsci/jenkins/pull/4936#pullrequestreview-490454698)._

File: core/src/main/resources/lib/form/radioBlock/radioBlock.js
Comment (daniel-beck) [reply core/src/main/resources/lib/form/radioBlock/radioBlock.js:23]: Looks like syntax error, should be

```suggestion
                n.style.position = "static";
```

Comment (romenrg) [reply core/src/main/resources/lib/form/radioBlock/radioBlock.js:23]: You are right, @daniel-beck . Thanks!
Although I'm testing it further and the position attribute change seems unnecessary here. Will post a new comment.

Comment (romenrg) [inline core/src/main/resources/lib/form/radioBlock/radioBlock.js:23]:
```suggestion
```

File: core/src/main/resources/lib/form/radioBlock/radioBlock.js
Comment (romenrg) [inline core/src/main/resources/lib/form/radioBlock/radioBlock.js:26]:
```suggestion
```

**Review by romenrg — CHANGES_REQUESTED** [review]
After doing some more local testing using Chrome, it seems changing the position is not needed to achieve the same effect (removing the "jump"). Seems that just setting visibility to "visible" or "hidden" accordingly does the trick.

If someone else can confirm they get the same behavior, then I would recommend to remove the position attribute changes and only set visibility, as I've attempted in the new suggestion I've made in this review.

File: core/src/main/resources/lib/form/radioBlock/radioBlock.js
Comment (benebsiny) [reply core/src/main/resources/lib/form/radioBlock/radioBlock.js:26]: Hello @romenrg , thanks for the suggestion! The reason why I set position to 'absolute' is that if there's something in the foldable block, there will reserve an empty space. See demo
o
So I change the position to 'absolute' to clear the empty space.

Comment (romenrg) [reply core/src/main/resources/lib/form/radioBlock/radioBlock.js:26]: I see, @benebsiny . Thank you for the explanation!
Then ignore my suggestion of removing the position attribute change. Just maybe consider the "static" one instead of the "", although that's not a big deal.

**Review by romenrg — APPROVED** [review]

File: core/src/main/resources/lib/form/radioBlock/radioBlock.js
Comment (benebsiny) [reply core/src/main/resources/lib/form/radioBlock/radioBlock.js:26]: I've update. Have a look at it.

Comment (romenrg) [reply core/src/main/resources/lib/form/radioBlock/radioBlock.js:26]: Awesome. Thanks!

**Comment by fqueiruga** [comment]
This PR has caused a regression in a non-public plugin https://issues.jenkins-ci.org/browse/JENKINS-64040. I'll try to see if I can get some screenshots.

**Pull request merged**
